src/imitation/algorithms/preference_comparisons.py

```diff
@@ -270,30 +275,35 @@ def sample(self, steps: int) -> Sequence[types.TrajectoryWithRew]:
 
         agent_trajs = _get_trajectories(agent_trajs, agent_steps)
 
-        exploration_trajs = []
+        trajectories = list(agent_trajs)
+
         if exploration_steps > 0:
             self.logger.log(f"Sampling {exploration_steps} exploratory transitions.")
             sample_until = rollout.make_sample_until(
                 min_timesteps=exploration_steps,
                 min_episodes=None,
             )
+            algo_venv = self.algorithm.get_env()
+            assert algo_venv is not None
             rollout.generate_trajectories(
                 policy=self.exploration_wrapper,
-                venv=self.algorithm.get_env(),
+                venv=algo_venv,
                 sample_until=sample_until,
-                # buffering_wrapper collects rollouts from a non-deterministic policy
+                # buffering_wrapper collects rollouts from a non-deterministic policy,
                 # so we do that here as well for consistency.
                 deterministic_policy=False,
             )
             exploration_trajs, _ = self.buffering_wrapper.pop_finished_trajectories()
             exploration_trajs = _get_trajectories(exploration_trajs, exploration_steps)
-        # We call _get_trajectories separately on agent_trajs and exploration_trajs
-        # and then just concatenate. This could mean we return slightly too many
-        # transitions, but it gets the proportion of exploratory and agent transitions
-        # roughly right.
-        return list(agent_trajs) + list(exploration_trajs)
-
-    @TrajectoryGenerator.logger.setter
+            # We call _get_trajectories separately on agent_trajs and exploration_trajs
+            # and then just concatenate. This could mean we return slightly too many
+            # transitions, but it gets the proportion of exploratory and agent
+            # transitions roughly right.
+            trajectories.extend(list(exploration_trajs))
+        return trajectories
+
+    # Type ignore due to https://github.com/python/mypy/issues/5936
+    @TrajectoryGenerator.logger.setter  # type: ignore[attr-defined]
```

Comment:
Thought about doing that, decided not to in the end, as I was hesitant since we might forget to remove that once that's fixed. But you're right that it's cleaner until then and even if that stays it's not necessarily bad style.